task-sdk/src/airflow/sdk/api/datamodels/_generated.py

```diff
@@ -245,6 +268,19 @@ class TerminalStateNonSuccess(str, Enum):
     FAIL_WITHOUT_RETRY = "fail_without_retry"
 
 
+class TriggerDAGRunPayload(BaseModel):
+    """
+    Schema for Trigger DAG Run API request.
+    """
+
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    logical_date: Annotated[datetime | None, Field(title="Logical Date")] = None
+    conf: Annotated[dict[str, Any] | None, Field(title="Conf")] = None
+    reset_dag_run: Annotated[bool | None, Field(title="Reset Dag Run")] = False
```

Comment:
(I know this is generated, but commenting here because)

Does `reset_dag_run` need to be optional, or would `: bool = False` be enough?

Reply:
Do we not also need the ability to specify a run_id? I thought that was possible somehow?

Reply:
The API Server has the same, but the auto-generation adds Optional.